Update documentation to include gen_dmrpp_side_car instructions

Mikejmnez opened the pull request:
THis PR updates the documentation, which is lacking some of the updates already in place for the generation of DMR++ files for HDF5 and HDF4.

See #10

Mikejmnez committed:
initial commit to update documentation

jgallagher59701 commented on DMRpp.adoc:
```diff
@@ -517,6 +517,30 @@ In order for a dataset to be Climate Forecast (CF) compatible it must contain th
 The Hyrax _hdf5_handler_ software, utilized by the _get_dmrpp_ application, can create this data from the dataset metadata.  The _get_dmrpp_ application places these generated data in a “sidecar” file for deployment with the source _hdf5/netcdf-4_ file.
 
 
+== [NEW] `gen_dmrpp_sidecar` for building +++ DMR++ +++ files HDF5 and HDF4 ==
```
There's a missing '_' in 'sidecar' It should be 'side_car'

Mikejmnez replied:
fixed

jgallagher59701 reviewed (changes requested):
Sorry, I didn't see this was a Draft PR...

Lets take a step back and think about the documentation a bit more.

Are the older command for hdf4 and hdf5 removed?

Mikejmnez committed:
fix typoe

Mikejmnez commented:
> Are the older command for hdf4 and hdf5 removed?

No they are not. Not yet at least. That is because the new function is only available on `snapshot`. I just added a new brief chapter for this new function.

If looks like the PR is building the documentation and pushing it to Github pages. So you can see what it looks like [on the official documentation](https://opendap.github.io/DMRpp-wiki/DMRpp.html#_new_gen_dmrpp_side_car_for_building_dmr_files_hdf5_and_hdf4) . You may need to clear you cache.

jgallagher59701 commented:
> > Are the older command for hdf4 and hdf5 removed?
> 
> No they are not. Not yet at least. That is because the new function is only available on `snapshot`. I just added a new brief chapter for this new function.
> 
> If looks like the PR is building the documentation and pushing it to Github pages. So you can see what it looks like [on the official documentation](https://opendap.github.io/DMRpp-wiki/DMRpp.html#_new_gen_dmrpp_side_car_for_building_dmr_files_hdf5_and_hdf4) . You may need to clear you cache.

I can see the new info. I'll look into the command a bit...

jgallagher59701 commented:
@Mikejmnez I have a ticket this sprint to update the documentation so people will know to use the new command. I'll also modify the .travis.yml file so that only the master/main branch is built and deployed. That way, I can hack/push as much as I want without altering the docs we present on the web.

jgallagher59701 committed:
Added a DMRpp macro

jgallagher59701 committed:
Update to newer builder - WIP

jgallagher59701 committed:
Document how to use gen_dmrpp_side_car

jgallagher59701 committed:
WIP

jgallagher59701 committed:
WIP

jgallagher59701 committed:
WIP

jgallagher59701 committed:
Major changes for gen_dmrpp_side_car

jgallagher59701 committed:
Updated the Introduction

jgallagher59701 committed:
Supported formats WIP

jgallagher59701 committed:
Organization and text. WIP

jgallagher59701 committed:
This version moves older text in comment blocks.

This is a draft that we should review in the group.

jgallagher59701 committed:
Edits complete except: Some Section references are broken. (med priority)

font-size and highlighting in the source listings is broken (low priority)

jgallagher59701 committed:
Minor edit regarding REST URLs

jgallagher59701 committed:
Figured out highlighting for code blocks

Use [source,shell, linenums,highlight=6-7]. However, this uses some horizontal space.

jgallagher59701 committed:
More linenums & highlight in the source listings

jgallagher59701 committed:
Added Why we did this paragraph

Mikejmnez commented:
@jgallagher59701 I added my reviews. And in addition to those:

* Are the old HDF4/HDF4 builders connected to the documentation somehow?

Links to these do not appear on the (local) build of the documentation. I think it is important to have them findable, for backwards compatibility. And state that on the test. You can add them as APPENDIXES at the bottom of the documentation.

jgallagher59701 commented:
> @jgallagher59701 I added my reviews. And in addition to those:
> 
> * Are the old HDF4/HDF4 builders connected to the documentation somehow?
> 
> Links to these do not appear on the (local) build of the documentation. I think it is important to have them findable, for backwards compatibility. And state that on the test. You can add them as APPENDIXES at the bottom of the documentation.

Yes, I can add those back in as appendices. They are in the doc as comments.

jgallagher59701 commented on DMRpp.adoc:
```diff
@@ -1,529 +1,416 @@
+The {DMRpp} is a metadata file that provides a fast and flexible way to access data stored in Amazon Web Services (AWS) Simple Storage Service (S3), or using any other service that supports HTTP.footnote:[The HTTP/S service must support the _Range_ header of HTTP/1.1. When using libcurl, both HTTP/S and the 'file:' protocols can be used.]
 
-. All chunks inside the `h5` file remain inside the file, i.e. each file retains its integrity. The location of the chunks inside the `h5` file are specified within the +++ DMR++ +++.
+The {DMRpp} encodes the location of data residing in a binary data file/object (e.g., an https://support.hdfgroup.org/documentation/HDF5/latest/[HDF5] file) so that it can be directly accessed, without the need for an intermediate Application Programmer Interface (API) library. The binary data objects may be on a local filesystem, or they may be accessible using HTTP in, for example, an S3 bucket ...
```
Fixed

jgallagher59701 commented on DMRpp.adoc:
```diff
@@ -1,529 +1,447 @@
+----
+
+[NOTE]
+If you want to use the latest version of the `gen+dmrpp_side_car` command, replace the version number in _opendap/hyrax:1.17.1-126_ with _snapshot_. Using _opendap/hyrax:snapshot_ will always get the most recent version of the software.
```
Hmm. This is not really about latest vs snapshot. I wanted to point out that they can use a specific version to get exactly a specific version (maybe because we told someone to do that in an email) or they can use snapshot to get the newest version of code. 

I used an explicit version in the document to ground the document - we can be sure people will get the decribed behavior with that version, not so with snapshot.

I'll try to clear up this note.

At this point, the old commands are not something they should use. We can add a note about old commands and backward (in)compatibility and send them off to the appendices.

jgallagher59701 commented on DMRpp.adoc:
```diff
@@ -1,529 +1,447 @@
---------------------------------------------------------------------------------------------------------------
+[source,sh]
+----
+% docker run -d -h hyrax -p 8080:8080 -v $DATA:/usr/share/hyrax --name=hyrax opendap/hyrax:1.17.1-126
```
I need to think about this one. My instinct is to keep the version numbers because it makes all the examples grounded. The issue is that we will have to update the version. The issue is also that examples using snapshot will gradually become untrue. So I have to think about this.

jgallagher59701 commented on DMRpp.adoc:
```diff
@@ -533,25 +451,24 @@ The +++ DMR++ +++ file(s) are XML files that contain a root `dap4:Dataset` eleme
+. Place the {DMRpp} files on the local disk inside the directory tree identified by the `BES.Catalog.catalog.RootDirectory` in the BES configuration.
 . Ensure that the Hyrax `AllowedHosts` list is configured to allow Hyrax to access those target URLs. This can be accomplished by adding new regex records to the `AllowedHosts` list in `/etc/bes/site.conf`, creating that file as need be.
-. If the data URLs require authentication to access then you'll need to configure Hyrax for that too.
+. If the data URLs require authentication to access, then you'll need to configure Hyrax for that too. See link:https://opendap.github.io/hyrax_guide/Master_Hyrax_Guide.html[The Hyrax Data Server Installation and Configuration Guide] for more information.
```
Done

jgallagher59701 commented on DMRpp.adoc:
```diff
@@ -1,529 +1,447 @@
+Click on your equivalent of the `3B42.20130111.09.7.HDF` link, subset, download, and open in Panoply or the equivalent.
 
 .Page view of the DAP _Data Request Form_ for subsetting the dataset.
 image::Hyrax-subsetting.png[width=650, height=400]
```
done

jgallagher59701 committed:
Made changes based on a review by Miguel.

jgallagher59701 committed:
Added linenum to a few source listings

jgallagher59701 committed:
I went through all the grammar suggestions.

jgallagher59701 committed:
Edits for -U

Also, moved the get_dmrpp_h4 text up and modified its description.
This is currently used while the older get_dmrpp is not.
